inc/shortcodes/class-vine.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class Vine extends Shortcode {
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => esc_html__( 'Vine', 'shortcake-bakery' ),
+			'listItemImage' => '<img width="100px" height="100px" style="padding-top:10px;" src="' . esc_url( SHORTCAKE_BAKERY_URL_ROOT . 'assets/images/svg/icon-vine.svg' ) . '" />',
+			'attrs'          => array(
+				array(
+					'label'        => esc_html__( 'URL', 'shortcake-bakery' ),
+					'attr'         => 'url',
+					'type'         => 'text',
+					'description'  => esc_html__( 'Full URL to the Vine', 'shortcake-bakery' ),
+				),
+			),
+		);
+	}
+
+	public static function reversal( $content ) {
+
+		$needle = '#<iframe[^>]+src="https?://vine\.co/v/([\w]+)(/[^"]+)?"[^>]+></iframe>#';
+		if ( preg_match_all( $needle, $content, $matches ) ) {
+			$replacements = array();
+			$shortcode_tag = self::get_shortcode_tag();
+			foreach ( $matches[0] as $key => $value ) {
+				$replacement_url = 'https://vine.co/v/' . $matches[1][ $key ];
```

Comment:
Same question here. If that's a pattern established elsewhere, I'm fine with it. Just curious what the rationale was..